File: Recursion/Queen/Queens2.py
def queens(n):
    solutions = []
    state = [None for _ in range(n)]

    def conflict(row, col):
        for j in range(col):
            if abs(state[j] - row) in (0, col-j):
                return True
        return False
        
    def put(col=0):
        for row in range(n):
            if not conflict(row, col):
                state[col] = row
                if col is n-1:
                    solutions.append(state.copy())
                else:
                    put(col+1)
    put(0)
    return solutions


for solution in queens(4):
    print(solution)

File: Recursion/Queen/Queens2.jl
function queens(n)
    solutions = []
    state = [0 for _ ∈ 1:n]
    function conflict(row, col)
        for c ∈ 1:col - 1
            if abs(state[c] - row) ∈ (0, col - c)
                return true
            end
        end
        return false
    end
    function put(col=1)
        for row ∈ 1:n
            if !conflict(row,col)
                state[col] = row
                if col == n
                    solutions = [solutions;[copy(state)]]
                else
                    put(col+1)
                end
            end
        end
    end
    put()
    return solutions
end

for solution ∈ queens(4)
    println(solution)
end